pandas/core/groupby/ops.py

```diff
@@ -717,6 +726,7 @@ def _aggregate_series_pure_python(self, obj, func):
             result[label] = res
 
         result = lib.maybe_convert_objects(result, try_float=0)
+        # TODO: try_cast back to EA?
```

Comment:
This is done in the code calling this (via `agg_series`):

https://github.com/pandas-dev/pandas/blob/82df98a04cea9ebb84df2a1271dc1a23f55212ba/pandas/core/groupby/groupby.py#L899-L905